pkgs/ffigen/lib/src/config_provider/config.dart

```diff
@@ -12,368 +14,694 @@ import 'config_types.dart';
-  final List<ImportedType> nativeTypeMappings;
+  static Declarations include(Set<String> names) => Declarations(
+    shouldInclude: (Declaration decl) => names.contains(decl.originalName),
+  );
+}
 
-  Map<String, ImportedType> get _nativeTypeMappings =>
-      Map<String, ImportedType>.fromEntries(
-        nativeTypeMappings.map(
-          (import) => MapEntry<String, ImportedType>(import.nativeType, import),
-        ),
-      );
+/// Configuration for globals.
+final class Globals extends Declarations {
+  const Globals({
+    super.rename,
+    super.renameMember,
+    super.shouldInclude,
+    super.shouldIncludeMember,
+    super.shouldIncludeSymbolAddress,
+  });
 
-  /// Extracted Doc comment type.
-  final CommentType commentType;
+  static const excludeAll = Globals(shouldInclude: _excludeAll);
 
-  /// Whether structs that are dependencies should be included.
-  final CompoundDependencies structDependencies;
+  static const includeAll = Globals(shouldInclude: _includeAll);
 
-  /// Whether unions that are dependencies should be included.
-  final CompoundDependencies unionDependencies;
+  static Globals include(Set<String> names) => Globals(
+    shouldInclude: (Declaration decl) => names.contains(decl.originalName),
+  );
+}
 
-  /// Whether, and how, to override struct packing for the given struct.
-  final PackingValue? Function(Declaration declaration) structPackingOverride;
+/// Configures how Macros are handled.
+final class Macros extends Declarations {
+  const Macros({
+    super.rename,
+    super.renameMember,
+    super.shouldInclude,
+    super.shouldIncludeMember,
+    super.shouldIncludeSymbolAddress,
+  });
 
-  static PackingValue? _structPackingOverrideDefault(Declaration declaration) =>
-      null;
+  static const excludeAll = Macros(shouldInclude: _excludeAll);
 
-  /// The module that the ObjC interface belongs to.
-  final String? Function(Declaration declaration) interfaceModule;
+  static const includeAll = Macros(shouldInclude: _includeAll);
 
-  static String? _interfaceModuleDefault(Declaration declaration) => null;
+  static Macros include(Set<String> names) => Macros(
+    shouldInclude: (Declaration decl) => names.contains(decl.originalName),
+  );
+}
 
-  /// The module that the ObjC protocol belongs to.
-  final String? Function(Declaration declaration) protocolModule;
+/// Configuration for Objective-C.
+final class ObjectiveC {
+  /// Declaration filters for Objective C interfaces.
+  final ObjCInterfaces interfaces;
 
-  static String? _protocolModuleDefault(Declaration declaration) => null;
+  /// Declaration filters for Objective C protocols.
+  final ObjCProtocols protocols;
 
-  /// Name of the wrapper class.
-  final String wrapperName;
+  /// Declaration filters for Objective C categories.
+  final ObjCCategories categories;
 
-  /// Doc comment for the wrapper class.
-  final String? wrapperDocComment;
+  /// Undocumented option that changes code generation for package:objective_c.
```

Comment:
"undocumented"? Isn't this comment documenting it?

Reply:
I presume it was undocumented in the YAML config. And then it made it into the
doc comments when drafting the first version of the Dart API.

But yes, we should probably see if we can manage to hide internal things.
(Though, `package:objective_c` is another package, so it cannot access internal
things.)